js/views/QueueTreeView.js

```diff
@@ -243,19 +243,19 @@ class QueueTreeView extends EventEmitter {
         if (maxDepth < 0) return;
         for (let i = 0; i <= maxDepth; i++) {
             const header = DomUtils.createElement('div', 'level-header', null, `Level ${i + 1}`);
-            this.levelHeadersContainerEl.appendChild(header);
+            this.levelHeadersContainerEl.append(header);
         }
     }
 
     _calculateMaxDepthOfFormattedTree(node) {
         if (!node) return -1;
-        let maxDepth = node.level !== undefined ? node.level : 0;
+        let maxDepth = node.level === undefined ? 0 : node.level;
         if (node.children) {
-            Object.values(node.children).forEach((child) => {
+            for (const child of Object.values(node.children)) {
                 if (child && !child.isDeleted) {
```

Comment:
if we have a 4 lvl tree and the 4th level only contains a deleted child this will work?